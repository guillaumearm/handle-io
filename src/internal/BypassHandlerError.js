const BypassHandlerError = function(...args) {
  const e = new Error(...args);
  this.e = e;
  this.message = e.message;
  this.stack = e.stack;
}

BypassHandlerError.prototype.toString = function() {
  return `${this.constructor.name}: ${this.e.message}`;
}

export default BypassHandlerError;
