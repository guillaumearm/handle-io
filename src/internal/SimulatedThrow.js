const SimulatedThrow = function(e) {
  this.e = e;
}

export default SimulatedThrow;
