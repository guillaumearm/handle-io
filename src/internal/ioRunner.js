const ioRunner = ({ f, args }) => f(...args);

export default ioRunner;
