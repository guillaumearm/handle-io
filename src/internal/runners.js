import isEqual from 'lodash.isequal';
import { stringify } from './utils';

export const createTestRunner = (mockedIOs = []) => {
  let mockIndex = 0;
  return io => {
    if (!mockedIOs[mockIndex]) {
      throw new Error('Mocked IOs should be exhaustive')
    }
    const [expectedIO, mockedRetValue] = mockedIOs[mockIndex];
    if (!isEqual(io.f, expectedIO.f)) {
      throw new Error(`Invalid IO#${mockIndex} function`)
    }
    if (!isEqual(io.args, expectedIO.args)) {
      const expectedArgs = stringify(expectedIO.args);
      const ioArgs = stringify(io.args);
      throw new Error(`Invalid IO#${mockIndex} function arguments: expected \n${expectedArgs}\nbut got \n${ioArgs}`)
    }
    mockIndex += 1;
    return mockedRetValue;
  }
};

export const ioRunner = ({ f, args }) => f(...args);
