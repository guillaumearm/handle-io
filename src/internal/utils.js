export const stringify = x => JSON.stringify(x, null, 2);
