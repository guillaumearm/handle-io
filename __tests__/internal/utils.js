import { stringify } from '../../src/internal/utils';

describe('handle-io/internal/utils', () => {
  describe('stringify', () => {
    test('JSON.stringify with padding 2', () => {
      const obj = { a: true, b: true, c: ['d', 'e', 'f'] };
      expect(stringify(obj)).toMatchSnapshot();
    });
  });
});
