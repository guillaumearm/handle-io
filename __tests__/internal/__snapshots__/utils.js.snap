// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle-io/internal/utils stringify JSON.stringify with padding 2 1`] = `
"{
  \\"a\\": true,
  \\"b\\": true,
  \\"c\\": [
    \\"d\\",
    \\"e\\",
    \\"f\\"
  ]
}"
`;
